### DISCUSSION TITLE
Ansi Terminal Request: _GVIDMODE command for changing video modes

### DISCUSSION THREAD

**Discussion by eightbitswide** [fdivitto/FabGL#284] [Ideas]
Could a _GVIDMODE num $ be added for changing the video resolution/color?

"1280x768, B&W",           // 0
 "1024x720, 4 Colors",      // 1
 "800x600, 8 Colors",       // 2
"720x520, 16 Colors",      // 3
 "640x480 73Hz, 16 C.",     // 4
 "640x480 60Hz, 16 C.",     // 5
"640x350, 16 Colors",      // 6
"512x384, 64 Colors",      // 7
 "400x300, 64 Colors",      // 8

**Comment by eightbitswide** [comment]
Cancel this request. I was reading all the terminal code in the library, before I completely read the entirety of the AnsiTerminal code.   Looking for the answer in the guts of the code, when it was staring me in the face the whole time.

**Reply by toml12953** [reply]
I'm not a C programmer. How would I set resolutions using escape codes from BASIC?

**Reply by johngalt01** [reply]
10 PRINT CHR$(27) +"_#512x384x64$"
    
    // Usable options:
    // "1280x768x2","1024x720x4","800x600x8","720x520x16","640x480@73x16"
    // "640x480@60x16","640x350x16","512x384x64","400x300x64"

**Comment by toml12953** [comment]
According to the source code, you need an R in the command:

 

// 'R': change resolution (for example: ESC + "_#R512x384x64$")

 

So wouldn’t the command be

 

10 PRINT CHR$(27) +"_#R512x384x64$"
